src/components/LogViewer/index.scss

```diff
@@ -0,0 +1,262 @@
+        &.text {
+          color: pink;
+        }
+
+        &.key {
+          color: red;
+        }
+
+        &.value {
+          color: blue;
+        }
+
+        &.invalid_pair {
+          text-decoration: underline;
+          text-decoration-color: red;
+        }
+      }
+    }
+
+    #log-query-actions {
+      grid-column: 1;
+      grid-row: 1;
+      justify-self: end;
+    }
+  }
+
+  #log-history {
+    overflow: auto;
+    padding: 10px 0px;
+    font-family: monospace;
+    font-size: 0.9em;
+    background-color: #f9f9f9;
+
+    #log-history-empty {
+      padding: 0px 10px;
+    }
+  }
+
+  #debug {
+    opacity: 0.2;
+    padding: 10px;
+    border-radius: 10px;
+  }
+
+  .log-entry {
+    display: grid;
+    grid-template-columns: fit-content(100%) fit-content(100%) 1fr fit-content(
+        100%
+      );
+    grid-template-rows: fit-content(100%) fit-content(100%);
+    grid-template-areas: 'timestamp level message expand';
+    grid-column-gap: 10px;
+    align-items: center;
+
+    &:first-child {
+      border-top: 1px solid #eee;
+    }
+
+    .log-timestamp {
+      grid-area: timestamp;
+      opacity: 0.5;
+    }
+
+    .log-level {
+      grid-area: level;
+
+      background-color: rgba(0, 0, 0, 0.1);
+      border-radius: 100%;
+      width: 20px;
+      height: 20px;
+      color: rgba(0, 0, 0, 1);
+      text-align: center;
+      padding: 4px;
+      font-weight: bold;
+      cursor: pointer;
+
+      &.warn {
+        background-color: orange;
+      }
+
+      &.user {
+        background-color: lightblue;
+      }
+
+      &:hover {
+        background-color: rgba(0, 0, 0, 0.15);
+      }
+    }
+
+    .log-message-single {
+      grid-column: 3 / span 1;
+      grid-row: 1;
+      line-height: 1.2em;
+      word-break: break-all;
+    }
+
+    .log-message-multi {
+      grid-column: 1 / span 4;
+      grid-row: 2;
+    }
+
+    .log-data-rich {
+      display: none;
+      visibility: hidden;
+    }
+
+    .log-toggle {
+      grid-area: expand;
+      background-color: #eee;
+      border-radius: 4px;
+      width: 30px;
+      height: 30px;
+      margin: 0px;
+      opacity: 1;
+      cursor: pointer;
+      display: grid;
+      appearance: unset;
+      align-content: center;
+      justify-content: center;
+      font-size: 1.2em;
+      font-weight: bold;
+      outline: none;
+      //
+      &:hover {
+        background-color: #ddd;
+      }
+
+      &:after {
+        display: grid;
+        width: 30px;
+        height: 30px;
+        align-content: center;
+        justify-content: center;
+      }
+
+      &:checked:after {
+        content: '-';
+      }
+
+      &:not(:checked):after {
+        content: '+';
+      }
+    }
+
+    .log-toggle:checked ~ .log-data-rich {
+      display: block;
+      visibility: visible;
+      grid-row: 2;
+      grid-column: 1 / span 4;
+    }
+
+    border-bottom: 1px solid #eee;
+    padding: 5px 10px;
+
+    pre {
+      margin: 10px 0px 0px 0px;
```

Comment:
in general all spacing and sizing should be multiples of 4px, but I will do a visual pass later

Reply:
sgtm